hopr-socks/src/main.rs

```diff
@@ -0,0 +1,203 @@
+use anyhow::Context;
+use fast_socks5::client::{Config as ClientConfig, Socks5Stream};
+use fast_socks5::server::{Authentication, Config as ServerConfig, SimpleUserPassword, Socks5Socket};
+use fast_socks5::Result;
+use log::{debug, error, info, warn};
+use std::future::Future;
+use std::sync::Arc;
+use structopt::StructOpt;
+use tokio::{
+    io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt},
+    net::TcpListener,
+    task,
+};
+
+/// # How to use it:
+///
+/// ## Run in server mode:
+///
+/// $ hopr-socks --host 127.0.0.1 --port 1337 server no-auth
+/// $ hopr-socks --host 127.0.0.1 --port 1337 server password --username admin --password password
```

Comment:
Regarding the command structure it'd be better to be able to run like this:
`hopr-socks run client --host localhost --port 12344 <other args>`

Reply:
Good point.